server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -543,11 +601,18 @@ public long getRestoreThrottleTimeInNanos() {
         return restoreRateLimitingTimeInNanos.count();
     }
 
+    protected void verificationThreadCheck() {
+        assert Thread.currentThread().getName().contains(ThreadPool.Names.SNAPSHOT) :
+            "Expected current thread [" + Thread.currentThread() + "] to be the snapshot thread.";
+    }
+
     @Override
     public String startVerification() {
+        verificationThreadCheck();
         try {
             if (isReadOnly()) {
-                // It's readonly - so there is not much we can do here to verify it
+                // It's readonly - so there is not much we can do here to verify it apart try to create blobStore()
+                blobStore();
```

Comment:
do you mean that comment has to be - `try the best - try to create blobStore on master` ?

Reply:
I think we need to extend verification so that it also works on read-only repos (and make it work in BWC-compatible fashion). Also, this should probably call `blobStore().blobContainer(basePath())`.
We can do that in a follow-up, but for now add a comment that this currently only checks that the repository is accessible on the master. // TODO: add repository verification for read-only repositories

Reply:
makes sense to me